relations/jf/src/deposit.rs

```diff
@@ -0,0 +1,139 @@
+use jf_relation::PlonkCircuit;
+
+use crate::{
+    note::{NoteGadget, NoteType, SourcedNote},
+    shielder_types::{Note, Nullifier, TokenAmount, TokenId, Trapdoor},
+    CircuitField, PlonkResult, PublicInput, Relation,
+};
+
+pub struct DepositRelation {
+    deposit_note: SourcedNote,
+}
+
+impl Default for DepositRelation {
+    fn default() -> Self {
+        Self::new(Default::default(), Default::default())
+    }
+}
+
+impl DepositRelation {
+    pub fn new(public: DepositPublicInput, private: DepositPrivateInput) -> Self {
+        Self {
+            deposit_note: SourcedNote {
+                note: public.note,
+                token_id: public.token_id,
+                token_amount: public.token_amount,
+                trapdoor: private.trapdoor,
+                nullifier: private.nullifier,
+                note_type: NoteType::Deposit,
+            },
+        }
+    }
+}
+
+#[derive(Default)]
```

Comment:
I find it weird that there exists a `Default` instance of `DepositPublicInput`. My understanding of the `Default` concept is that it exists for things that it's correct to supply that default instance, that it makes sense in some contexts. I can't imagine a case where there's a default public input to a relation, even more if that input is token ID, amount, note etc. A default instance of `Note` is probably invalid in any context, other than for generating keys.